frontend/src/components/pages/RegistrantExperience/Waiver/index.tsx

```diff
@@ -51,16 +58,63 @@ const boldButtonTextStyles = {
 const marginBetweenSections = { sm: 7, md: 8, lg: 10 };
 
 interface WaiverPageProps {
+  nextBtnRef: React.RefObject<HTMLButtonElement>;
   waiverInterface: WaiverInterface;
   waiverDispatch: React.Dispatch<WaiverReducerDispatch>;
   campName: string;
 }
 
 const WaiverPage = ({
+  nextBtnRef,
   waiverInterface,
   waiverDispatch,
   campName,
 }: WaiverPageProps): React.ReactElement => {
+  const [isNameInvalid, setIsNameInvalid] = useState<boolean>(false);
+  const [isDateInvalid, setIsDateInvalid] = useState<boolean>(false);
+  const [
+    isRequiredClauseInvalid,
+    setIsRequriedClauseInvalid,
+  ] = useState<boolean>(false);
+  const [isOptionalClausesInvalid, setIsOptionalClausesInvalid] = useState<
+    boolean[]
+  >([]);
+
+  useEffect(() => {
+    // update the size of the isOptionalClausesInvalid accordingly if clauses are added or deleted
+    setIsOptionalClausesInvalid(
+      new Array(waiverInterface.optionalClauses.length).fill(false),
+    );
+  }, [waiverInterface.optionalClauses.length]);
+
+  useEffect(() => {
+    let nextBtnRefValue: HTMLButtonElement; // Reference to the next step button
+    const updateFormErrorMsgs = () => {
+      if (!checkName(waiverInterface.name)) setIsNameInvalid(true);
+      if (!checkDate(waiverInterface.date)) setIsDateInvalid(true);
+      if (!checkRequiredClauses(waiverInterface.agreedRequiredClauses))
+        setIsRequriedClauseInvalid(true);
```

Comment:
can we add curly braces around this if check? It's a little confusing as to what's part of the if block etc.

Reply:
Yep!

Reply:
Ok it seems that lint doesn't like that style. I will put the brackets instead as you suggested.